docs/DeveloperGuide.md

```diff
@@ -1,38 +1,230 @@
+2. Help messages that details the usage of the possible commands available in the program (`PrintHelpMessage` class)
+3. Error messages that trigger when a user's input is incorrect and provides an explanation to the
+   user. (`ErrorMessages` class)
+
+### Storage Component
+
+API: [```StorageHandler.java```](https://github.com/AY2223S2-CS2113-W13-2/tp/blob/master/src/main/java/seedu/duke/storage/StorageHandler.java)
+
+The *Storage* component handles the reading and writing of user data to and from the local hard disk in the form of
+a json file.
+
+Key Aspects:
+
+* Handles the serialization and deserialization of user data into a json file using
+  the [Gson Library](https://github.com/google/gson)
+* Handles the creation of user data file when previous one is missing or corrupted
+* Handles the loading of user data and plans upon start of the program
+
+The class diagram as shown in *Figure 1.3* illustrates the structure of the different classes in Storage.
+<div align="center">
+<img src="UML/Images/Storage.png"/>
+<p>
+Figure 1.3
+</p>
+</div>
+
+The StorageHandler interacts with the other classes as shown in the *Sequence Diagram*
+where it shows how the StorageHandler loads the local user data json file as well as the user plans json file upon the 
+resumption of the program.
+<div align="center">
+<img src="UML/Images/LoadingUserCareerData.png"/>
+<p>
+Figure 1.4
+</p>
+</div>
+
+### Command Handler Component
+
+The command handler component consists of multiple states that the program can be at any time. 
+It controls where the user input is being processed.
+
+#### Key Aspects: 
+
+* Handles when there is an exercise ongoing, giving access to finishing and hence saving finished exercises, and keeping track of what exercises there are.
+* Handles when there is no exercise ongoing, granting access to generating new exercises with different confines.
+
+<div align="center">
+<img src="UML/Images/CommandHandler.png"/>
+<p>
+Figure 1.5
+</p>
+</div>
+
+### ExerciseSessionCommandHandler
+The exercise Command Handler is engaged when the user has an ongoing workout.
+Upon completion of the workout, the user can save their workout if completed, such
+that the completed workout is logged and will be saved for future reference.
+
+To manage this exists the ```ExerciseStateHandler```, which allows for saving, starting
+ending, cancelling workouts.
+<div align="center">
+<img src="UML/Images/CommandHandler.png"/>
+<p>
+Figure 1.6
+</p>
+</div>
+
+### Exceptions
+
+Accounts for the different scenarios that may trigger an error during user's interactions with the program
+
+#### Error Message Handling
+Enumeration: [```ErrorMessages.java```]
+All error messages are stored in the ErrorMessage enumeration for easy access across different classes that could run into similar exceptions.
+<img src="UML/Images/ErrorMessagesEnum.png"/>
+<div align="center">
+<p>
+Figure 1.7
+</p>
+</div>

```

Comment:
![image](https://user-images.githubusercontent.com/88542282/228426090-29c483a8-6d7d-40e6-baae-0aefcf46376a.png)
Could consider including the enumeration values for `ErrorMessages`